core/src/main/java/com/netflix/conductor/domain/WorkflowDO.java

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.netflix.conductor.domain;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.netflix.conductor.common.metadata.workflow.WorkflowDef;
+
+import com.google.common.base.Preconditions;
+
+import static com.netflix.conductor.domain.WorkflowStatusDO.RUNNING;
+
+public class WorkflowDO {
```

Comment:
I am wondering should we use a different name convention for xxxDO? I understand it's short for domain object, but isn't that intuitive when first hearing it.

Reply:
changed to WorkflowModel